File: Examples/SimpleTransformix.cxx
// This one header will include all SimpleITK filters and external
// objects.
#include <SimpleITK.h>
#include <iostream>
#include <stdlib.h>

namespace sitk = itk::simple;

int main ( int argc, char* argv[] ) {

  if ( argc < 4 ) {
    std::cerr << "Usage: " << argv[0] << " <fixedImage> <movingImage> <inputImage> <parameterFile> <outputImage>\n";
    return 1;
  }

  // Make transform
  sitk::SimpleElastix elastix;
  sitk::ImageFileReader reader;
  reader.SetFileName( std::string( argv[1] ) );
  elastix.SetFixedImage( reader.Execute() );
  reader.SetFileName( std::string( argv[2] ) );
  elastix.SetMovingImage( reader.Execute() );
  elastix.SetParameterMap( sitk::ReadParameterFile( std::string( argv[3] ) ) );
  elastix.LogToConsoleOn();
  elastix.Execute();

  // Instantiate transformix
  sitk::SimpleTransformix transformix;

  // Read input
  reader.SetFileName( std::string( argv[4] ) );
  transformix.SetInputImage( reader.Execute() );
  transformix.SetTransformParameterMap( elastix.GetTransformParameterMap() );

  // Perform warp
  transformix.LogToConsoleOn();
  transformix.Execute();

  // Write result image
  sitk::ImageFileWriter writer;
  writer.SetFileName( std::string( argv[5] ) );
  writer.Execute( transformix.GetResultImage() );

  return 0;
}

File: Examples/Java/SimpleTransformix.java
import org.itk.simple.*;

class SimpleTransformix {
 
  public static void main(String argv[]) {

    // If you get "no SimpleITKJava in java.library.path" point java.library.path to lib/libSimpeITKJava.jnilib

    if ( argv.length < 4 ) {
      System.out.println("Usage: java -cp SimpleTransformix.jar SimpleTransformix <fixedImage> <movingImage> <parameterFile> <inputImage> <output>");
      return;
    }

    // Make transform
    ImageFileReader reader = new ImageFileReader();
    reader.setFileName(argv[0]);
    ElastixImageFilter elastixImageFilter = new ElastixImageFilter();
    elastixImageFilter.setFixedImage(reader.execute());
    reader.setFileName(argv[1]);
    elastixImageFilter.setMovingImage(reader.execute());
    elastixImageFilter.setParameterMap(elastixImageFilter.readParameterFile(argv[2]));
    elastixImageFilter.logToConsoleOn();
    elastixImageFilter.execute();

    // Instantiate transformix
    TransformixImageFilter transformixImageFilter = new TransformixImageFilter();

    // Read input
    reader.setFileName(argv[3]);
    transformixImageFilter.setMovingImage(reader.execute());
    transformixImageFilter.setTransformParameterMap(elastixImageFilter.getTransformParameterMap());

    // Perform warp
    transformixImageFilter.logToConsoleOn();
    transformixImageFilter.execute();

    // Write result image
    ImageFileWriter writer = new ImageFileWriter();
    writer.setFileName(argv[4]);
    writer.execute(transformixImageFilter.getResultImage()); 

  }

}
